src/build_meta.rs

```diff
@@ -29,15 +30,94 @@ pub enum BuildHistoryRetrieval {
     None,
 }
 
+#[derive(Debug)]
+pub struct BuildMetaWaitOptions {
+    pub sleep_on_empty: Duration,
+    pub max_empty_retries: usize,
+}
+
+impl Default for BuildMetaWaitOptions {
+    fn default() -> Self {
+        Self {
+            sleep_on_empty: Duration::from_secs(15),
+            max_empty_retries: 16,
```

Comment:
does it make sens for there to be a max?

Reply:
From my understanding, it's entirely possible for this to to happen during "normal" operation due to other issues, it's just rare? So having some sort of cap is useful to make sure it wouldn't just spin forever if no one notices.

Reply:
Dropped the max limits.